includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2663,23 +2664,14 @@ public function capture_charge( $order, $include_level3 = true ) {
 		$http_code                = null;
 
 		try {
-			$intent_id = $order->get_transaction_id();
-
-			$request = Get_Intention::create( $intent_id );
-			$intent  = $request->send( 'wcpay_get_intent_request', $order );
-
-			$payment_type = $this->is_payment_recurring( $order->get_id() ) ? Payment_Type::RECURRING() : Payment_Type::SINGLE();
-
-			$metadata_from_intent = $intent->get_metadata(); // mobile app may have set metadata.
-			$metadata_from_order  = $this->get_metadata_from_order( $order, $payment_type );
-			$merged_metadata      = array_merge( (array) $metadata_from_order, (array) $metadata_from_intent ); // prioritize metadata from mobile app.
-
-			$wcpay_request = Update_Intention::create( $intent_id );
-			$wcpay_request->set_metadata( $merged_metadata );
-			$wcpay_request->send( 'wcpay_prepare_intention_for_capture', $order );
+			$intent_id           = $order->get_transaction_id();
+			$payment_type        = $this->is_payment_recurring( $order->get_id() ) ? Payment_Type::RECURRING() : Payment_Type::SINGLE();
```

Comment:
`Payment_Type::RECURRING() : Payment_Type::SINGLE();` - we should not call the constants as functions. This will generate an error.
`Payment_Type::RECURRING : Payment_Type::SINGLE;`

Reply:
Great catch! Fixed in https://github.com/Automattic/woocommerce-payments/pull/6408/commits/bc2f2a254c01d2a2ddcafa41e82e68ea752564c0.

Reply:
@dpaun1985 I had to revert this fix in https://github.com/Automattic/woocommerce-payments/pull/6408/commits/19184ee642603293f8bc58b064821df5d18598c5 because it was resulting in a psalm error in `WC_Payment_Gateway_WCPay::get_metadata_from_order` which expects payment type as `Payment_Type` instead of string. I cannot change the argument type in this method as it's being used in other places, like this:

https://github.com/Automattic/woocommerce-payments/blob/19184ee642603293f8bc58b064821df5d18598c5/includes/class-wc-payment-gateway-wcpay.php#L1125